leapp-report-analyzer.py

```diff
@@ -0,0 +1,199 @@
+import argparse
+import glob
+import json
+import os
+import sys
+from collections import OrderedDict, defaultdict
+
+# Check if Python version is 3.6 or higher (cgi.escape is deprecated/removed for newer versions)
+if sys.version_info.major >= 3 and sys.version_info.minor >= 6:
+    from html import escape
+else:
+    from cgi import escape
+
+SEVERITY_ORDER = ["inhibitor", "high", "medium", "low", "info"]
+HTML_TABLE_OPTIONS = "border='1' cellpadding='5' cellspacing='0'"
+
+
+def _create_grouped_data_structure():
+    # JSON structure: {severity: [key: {title: str, {hostnames: [], remediations: {}}]}
+    return defaultdict(lambda: defaultdict(lambda: {"title": "", "hostnames": [], "remediations": []}))
+
+
+def escape_html(text):
+    return escape(text, quote=True)
+
+
+def main():
```

Comment:
My preference would be to put main function to the bottom, it feels more natural to me to have necessary functions first and then function that uses them (in this case main).

Reply:
I disagree. I like top-down approach for structuring code. Most important first so that the code unwraps in a natural way to the bottom, not going down and than back up, each to their own I guess.

I only placed some helper functions at the top which are not really that much related to other functions.

Is this okay with you?

Reply:
yeah I would put the functions into some `helper.py` file since they seem to have different purpose than other functions (but we want to keep it all in one file), but I see your point about consistency, so will put main as top.

Added.

Reply:
alright, nice